Source/WebCore/dom/Document.cpp

```diff
@@ -9710,10 +9710,13 @@ void Document::applyContentDispositionAttachmentSandbox()
     ASSERT(shouldEnforceContentDispositionAttachmentSandbox());
 
     setReferrerPolicy(ReferrerPolicy::NoReferrer);
-    if (!isMediaDocument())
-        enforceSandboxFlags(SandboxFlags::all());
-    else
+
+    if (isMediaDocument())
         enforceSandboxFlags(SandboxFlag::Origin);
+    else if (isPluginDocument())
+        enforceSandboxFlags(SandboxFlags::all() - SandboxFlag::Plugins);
```

Comment:
This looks weird in that read out loud, this code says "If it's a plugin document, remove the plugin sandbox flag." Is there a way to make this obviously correct when you read it?

Reply:
I was thinking of a way to alleviate the counterintuitiveness for readers, but the best I could do was an explanatory comment:

```cpp
    auto sandboxFlags = [this] -> SandboxFlags {
        if (isMediaDocument())
            return SandboxFlag::Origin;
        if (isPluginDocument()) {
            // This reads counterintuitively, but for PluginDocuments, we want all other
            // sandbox restrictions to still apply, inspite of the fact that the sandbox
            // should _not_ restrict plugin instantiation.
            SandboxFlags::all() - SandboxFlag::Plugins;
        }
        return SandboxFlags::all();
    }();
```